src/plugins/discover/public/components/data_types/logs/service_name_badge_cell.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiText } from '@elastic/eui';
+import type { AgentName } from '@kbn/elastic-agent-utils';
+import { dynamic } from '@kbn/shared-ux-utility';
+import type { DataGridCellValueElementProps } from '@kbn/unified-data-table';
+import React from 'react';
+import { AGENT_NAME_FIELD } from '../../../../common/data_types/logs/constants';
+
+const dataTestSubj = 'serviceNameBadgeCell';
+const AgentIcon = dynamic(() => import('@kbn/custom-icons/src/components/agent_icon'));
+
+export const getServiceNameBadgeCell =
+  (serviceNameField: string) => (props: DataGridCellValueElementProps) => {
+    const serviceNameValue = props.row.flattened[serviceNameField];
+    const agentNameValue = props.row.flattened[AGENT_NAME_FIELD];
+
+    if (!serviceNameValue) {
+      return null;
+    }
+
+    const agentName = (
+      Array.isArray(agentNameValue) ? agentNameValue[0] : agentNameValue
+    ) as AgentName;
+
+    return (
+      <EuiFlexGroup gutterSize="xs" data-test-subj={`${dataTestSubj}-${agentName || 'unknown'}`}>
```

Comment:
```suggestion
      <EuiFlexGroup gutterSize="xs" responsive={false} alignItems="center" data-test-subj={`${dataTestSubj}-${agentName || 'unknown'}`}>
```
Currently on small screen sizes, the icon and text are shown on separate lines. I'd recommend adding `responsive={false}` to always keep them on the same line.

Also I think we should avoid using `<EuiText size="xs">` below since it overrides the text size when the data grid density is set to `Normal` or `Expanded`. Instead we could add `alignItems="center"` to the flex group to ensure the icon and text are always aligned.